cli/admin/src/main/java/io/pravega/cli/admin/segmentstore/tableSegment/GetTableSegmentEntryCommand.java

```diff
@@ -0,0 +1,84 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.pravega.cli.admin.segmentstore.tableSegment;
+
+import io.pravega.cli.admin.CommandArgs;
+import io.pravega.cli.admin.utils.AdminSegmentHelper;
+import io.pravega.client.tables.impl.TableSegmentEntry;
+import io.pravega.client.tables.impl.TableSegmentKey;
+import io.pravega.common.util.ByteArraySegment;
+import io.pravega.shared.protocol.netty.PravegaNodeUri;
+import lombok.Cleanup;
+import org.apache.curator.framework.CuratorFramework;
+
+import java.nio.ByteBuffer;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.CompletableFuture;
+
+import static io.pravega.cli.admin.serializers.AbstractSerializer.parseStringData;
+
+public class GetTableSegmentEntryCommand extends TableSegmentCommand {
+
+    /**
+     * Creates a new instance of the GetTableSegmentEntryCommand.
+     *
+     * @param args The arguments for the command.
+     */
+    public GetTableSegmentEntryCommand(CommandArgs args) {
+        super(args);
+    }
+
+    @Override
+    public void execute() {
+        ensureArgCount(3);
+        ensureSerializersExist();
+
+        final String fullyQualifiedTableSegmentName = getArg(0);
+        final String key = getArg(1);
+        final String segmentStoreHost = getArg(2);
+        @Cleanup
+        CuratorFramework zkClient = createZKClient();
+        @Cleanup
+        AdminSegmentHelper adminSegmentHelper = instantiateAdminSegmentHelper(zkClient);
+
+        ByteArraySegment serializedKey = new ByteArraySegment(getCommandArgs().getState().getKeySerializer().serialize(key));
+
+        CompletableFuture<List<TableSegmentEntry>> reply = adminSegmentHelper.readTable(fullyQualifiedTableSegmentName,
+                new PravegaNodeUri(segmentStoreHost, getServiceConfig().getAdminGatewayPort()),
+                Collections.singletonList(TableSegmentKey.unversioned(serializedKey.getCopy())),
+                super.authHelper.retrieveMasterToken(), 0L);
+
+        ByteBuffer serializedValue = getByteBuffer(reply.join().get(0).getValue());
+        String value = getCommandArgs().getState().getValueSerializer().deserialize(serializedValue);
+        output("For the given key: %s", key);
+        userFriendlyOutput(value);
+    }
+
+    public static CommandDescriptor descriptor() {
+        return new CommandDescriptor(COMPONENT, "get", "Get the value for the given key in the table.",
+                new ArgDescriptor("qualified-table-segment-name", "Fully qualified name of the table segment to get info from."),
+                new ArgDescriptor("key", "The key to be queried."),
+                new ArgDescriptor("segmentstore-endpoint", "Address of the Segment Store we want to send this request."));
+    }
+
+    private void userFriendlyOutput(String data) {
```

Comment:
I am not sure how this thing works with complex data structures.

Example is SLTS metadata where key is simple string but the data is a complex object of type [TransactionData](https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/BaseMetadataStore.java#L926) that contains embedded object of type  [SegmentMetadata](https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/SegmentMetadata.java)  or [ChunkMetadata](https://github.com/pravega/pravega/blob/master/segmentstore/storage/src/main/java/io/pravega/segmentstore/storage/metadata/ChunkMetadata.java)

Reply:
We get the binary value from the wire call and deserialize it and convert it back into that `key1=value1;key2=value2;...`. `userFriendlyOutput` then prints this string into a more readable format with newlines and such.

Reply:
It think the point here is that actual value is a complex object containing other object
Something like this

```
TransactionData {
version = 1234, 
value =  SegmentMetadata{
                name = "mysegment",
                status = 15
                 length = 7890
                 chunkcount = 19
                ....
                ....
             }
}
```

or 
TransactionData {
version = 5678, 
value =  ChunkMetadata{
                name = "mysegment.E-333-O-8888.BBFF-AAA",
                 length = 50
                 status = 1
                 nextChunk= "mysegment.E-333-O-9999.EF56-DCE",
                ....
                ....
             }
}

so my question was - how do we handle complex objects like that

Reply:
Based on the fields provided, we identify which SLTS metadata is being dealt with. The serializer identifies and then generates the required object.